packages/flutter/test/widgets/default_text_editing_shortcuts_test.dart

```diff
@@ -486,6 +486,185 @@ void main() {
       expect(state.lastIntent, isA<ExtendSelectionToLineBreakIntent>());
     }, variant: macOSOnly);
   }, skip: kIsWeb); // [intended] specific tests target non-web.
+
+  group('Linux does accept numpad shortcuts', () {
+    testWidgets('when numlock is locked', (WidgetTester tester) async {
+      final FocusNode editable = FocusNode();
+      addTearDown(editable.dispose);
+      final FocusNode spy = FocusNode();
+      addTearDown(spy.dispose);
+
+      await tester.pumpWidget(
+        buildSpyAboveEditableText(
+          editableFocusNode: editable,
+          spyFocusNode: spy,
+        ),
+      );
+      spy.requestFocus();
+      await tester.pump();
+      final ActionSpyState state = tester.state<ActionSpyState>(find.byType(ActionSpy));
+
+      // Lock NumLock.
+      await tester.sendKeyEvent(LogicalKeyboardKey.numLock);
+      expect(HardwareKeyboard.instance.lockModesEnabled.contains(KeyboardLockMode.numLock), isTrue);
```

Comment:
It looks like this gets automatically reset after the test?

Reply:
Yes it is. I first wanted to reset it but my thinking was that it might hide a bug in test bindings if one day the keyboard state reset no more works as expected.